### PR TITLE
Documenting the project setup guide #7

### PR DESCRIPTION
**Pull request by rajitha1998**
Fixes - #7  
## Changes proposed in the pull request
  * Documenting the project setup guide
## Impact
 * Read me file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,111 @@
-# MatEditor
+# Mat Editor
 
-This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 7.0.5.
+Mat Editor is a free, open source WYSIWYG editor built for the modern web which is developed and maintained by [Maturify](https://www.maturify.com/). It's a npm library which can be used on **Angular** based web applications.
```
Comment (Anuranie) [inline README.md]: Write what this abbreviation means please --> WYSIWYG

File: README.md [hunk 2]
```diff
@@ -1,27 +1,111 @@
...
+3.  Installing relevant packages 
+
+    Navigate to project directory and run:
+
+     - `git checkout dev` 
+     - `npm i` 
+
+4.  Building the Mat-editor
+
+* The repository contains of two main projects which is the mat-editor library and the demo which is used to test the library.whenever you make a change in the library make sure that you build the library before testing, unless the imported library in the demo won't be updated
```
Comment (Anuranie) [inline README.md]: 'library. Whenever'......

'change in the library make sure to build the library before testing. If not, the imported library in the demo won't be updated.'......
Add a full stop at the end of the sentence

File: README.md [hunk 3]
```diff
@@ -1,27 +1,111 @@
...
+*  Building the library > `ng build mat-editor` 
+*  Testing the library in demo > `ng serve` 
+
+## Publishing the Library(Only for maturify members)
```
Comment (Anuranie) [inline README.md]: Add a space before the bracketed text

Comment (Anuranie) [reply README.md]: Also make sure you use capital 'm' when mentioning 'Maturify'

File: README.md [hunk 4]
```diff
@@ -1,27 +1,111 @@
...
+Please follow the the following steps in order:
+
+* Sign in using > `npm login` 
+* Check the current user > `npm whoami` 
+* Upgrade the version number in `package.json ` (this is mandatory)
+* Build the library > `ng build mat-editor` 
+* Pack the npm package > `npm run build_lib && npm run npm_pack` 
+* Publish the library > `npm publish ./dist/mat-editor/mat-editor-x.x.x.tgz` (x.x.x represents the version)
+ 
+From the above commands the packed library in the `dist` folder will be published
```
Comment (Anuranie) [inline README.md]: Add a full stop at the end

**Review by Anuranie — APPROVED** [review]

**Review by psnmissaka — APPROVED** [review]

**Pull request merged**